controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -232,6 +246,42 @@ func (ew *EndpointsWatcher) Unsubscribe(id ServiceID, port Port, hostname string
 	sp.unsubscribe(port, hostname, listener)
 }
 
+// Stop will terminate an EndpointsWatcher by shutting down its informers. It
+// uses the write half of the channel used to sync the informers to signal
+// shutdown. It additionally de-registers any event handlers used by its
+// informers.
+func (ew *EndpointsWatcher) Stop(stopCh chan<- struct{}) {
```

Comment:
having this function take a channel and then just send a message on the channel seems like needless coupling and indirection?  `Stop` doesn't actually need the channel for anything.  The call sites could just send on the channel instead of passing it into this function.

Reply:
Following this train of thought, this function would only unregister the event handlers so it might be good to rename the function so that it doesn't suggest that this shuts down the informers.